openqabot/types/incidents.py

```diff
@@ -96,35 +96,37 @@ def __call__(
                             inc.id,
                         )
                         continue
+
+                    if inc.staging:
+                        continue
+
+                    flavor_settings = self.settings.copy()
+                    if "flavor_settings" in data:
+                        flavor_settings.update(data["flavor_settings"])
+
+                    if "packages" in data:
+                        if not inc.contains_package(data["packages"]):
+                            continue
+
+                    if "excluded_packages" in data:
+                        if inc.contains_package(data["excluded_packages"]):
+                            continue
+
                     full_post: Dict[str, Any] = {}
                     full_post["api"] = "api/incident_settings"
                     full_post["qem"] = {}
                     full_post["openqa"] = {}
-                    full_post["openqa"].update(self.settings)
```

Comment:
IMHO all changes what you have done above and below may be replaced by just TWO lines added here ( and we NOT removing this one too ofc. ) 

```
 if "flavor_settings" in data:
                 full_post["openqa"].update(data["flavor_settings"])
```

to be on safe side we may add verification something like : 
```
if "VERSION" in data["flavor_settings"] or "DISTRI" in data["flavor_settings"]:
                    raise ValueError("Attempt to redifine VERSION/DISTRI on flavor level")
```

Reply:
I'm ok if it pass the UT I created. I'm giving it a try

Reply:
@asmorodskyi can we please further elaborate idea in
```
if "VERSION" in data["flavor_settings"] or "DISTRI" in data["flavor_settings"]:
                    raise ValueError("Attempt to redifine VERSION/DISTRI on flavor level")
```

There's no other `raise` in that function. Should we do it and stop everything else, every other `flavors`? Can we just `continue` and skip scheduling  `POST` for the "malicious flavor"? Can we simple remove any `DISTRI` and `VERSION` from `flavor_settings` before to merge it to the `full_post["openqa"]` ?